src/Functions/FunctionsConversion.h

```diff
@@ -2825,13 +2825,13 @@ struct ToNumberMonotonicity
         /// (Enum has separate case, because it is different data type)
         if (checkAndGetDataType<DataTypeNumber<T>>(&type) ||
             checkAndGetDataType<DataTypeEnum<T>>(&type))
-            return { .is_monotonic = true, .is_always_monotonic = true };
+            return { .is_monotonic = true, .is_always_monotonic = true, .is_strict = true };
 
         /// Float cases.
 
         /// When converting to Float, the conversion is always monotonic.
         if constexpr (is_floating_point<T>)
-            return { .is_monotonic = true, .is_always_monotonic = true };
+            return { .is_monotonic = true, .is_always_monotonic = true, .is_strict = true };
```

Comment:
Not strictly monotonic:
```sql
SELECT
    toFloat32(1000000000000000000 AS x),
    toFloat32(x + 1)

   ┌────────toFloat32(x)─┬─toFloat32(plus(x, 1))─┐
1. │ 1000000000000000000 │   1000000000000000000 │
   └─────────────────────┴───────────────────────┘
```

Int to float is strictly monotonic when the int is within the corresponding float's mantissa range (51 bits for Float64?), can check that using `left` and `right`. Or just leave `is_strict == false`.

Reply:
Ok, fixed.

Reply:
> Not strictly monotonic:
> 
> ```sql
> SELECT
>     toFloat32(1000000000000000000 AS x),
>     toFloat32(x + 1)
> 
>    ┌────────toFloat32(x)─┬─toFloat32(plus(x, 1))─┐
> 1. │ 1000000000000000000 │   1000000000000000000 │
>    └─────────────────────┴───────────────────────┘
> ```
> 
> Int to float is strictly monotonic when the int is within the corresponding float's mantissa range (e.g. 51 bits for Float64, IIRC), can check that using `left` and `right`. Or just leave `is_strict == false`.

Ok, fixed. Thanks.